bug: http_post_form curl command missing -f flag masks actual server errors

## Description

The `http_post_form` function in the sandbox agent uses curl without the `-f` (--fail) flag, causing HTTP errors to be masked as "Invalid JSON response" instead of showing the actual error.

## Problem

**File:** `turbo/apps/web/src/lib/e2b/scripts/lib/http_client.py.ts` (lines 104-110)

```python
curl_cmd = [
    "curl", "-X", "POST", url,
    "-H", f"Authorization: Bearer {API_TOKEN}",
    "--connect-timeout", str(HTTP_CONNECT_TIMEOUT),
    "--max-time", str(HTTP_MAX_TIME_UPLOAD),
    "--silent"
]
```

**Issue:** Without `-f` flag, curl returns exit code 0 even when server returns HTTP 4xx/5xx errors. The HTML error page is then parsed as JSON, causing `JSONDecodeError` logged as "Invalid JSON response".

**Contrast with `http_download`** (line 178-181): Correctly uses `curl -fsSL` which includes the `-f` flag.

## Actual Error Log

```
[WARN] HTTP POST form failed (attempt 1/3): Invalid JSON response
[WARN] HTTP POST form failed (attempt 2/3): Invalid JSON response
[WARN] HTTP POST form failed (attempt 3/3): Invalid JSON response
[ERROR] HTTP POST form failed after 3 attempts to https://www.vm0.ai/api/webhooks/agent/storages
```

The actual server error (timeout, 502, 500, etc.) is completely hidden.

## Expected Behavior

- curl should fail with non-zero exit code on HTTP errors
- Actual HTTP status code should be logged
- Raw response body should be logged on failure for debugging

## Suggested Fix

1. Add `-f` flag to curl command
2. Optionally use `-w "%{http_code}"` to capture and log HTTP status code
3. Log raw response body when JSON parsing fails

```python
curl_cmd = [
    "curl", "-f", "-X", "POST", url,  # Add -f flag
    "-H", f"Authorization: Bearer {API_TOKEN}",
    "--connect-timeout", str(HTTP_CONNECT_TIMEOUT),
    "--max-time", str(HTTP_MAX_TIME_UPLOAD),
    "--silent",
    "-w", "\n%{http_code}"  # Optional: append HTTP status code
]
```

## Impact

- **Severity:** Medium (debugging/observability issue)
- **User impact:** Makes debugging checkpoint failures very difficult
- **Related:** #588

--- a/turbo/apps/web/src/lib/e2b/scripts/lib/http_client.py.ts
+++ b/turbo/apps/web/src/lib/e2b/scripts/lib/http_client.py.ts
@@ -102,8 +102,9 @@ def http_post_form(
         log_debug(f"HTTP POST form attempt {attempt}/{max_retries} to {url}")
 
         # Build curl command
+        # -f flag makes curl return non-zero exit code on HTTP 4xx/5xx errors
         curl_cmd = [
-            "curl", "-X", "POST", url,
+            "curl", "-f", "-X", "POST", url,
             "-H", f"Authorization: Bearer {API_TOKEN}",
             "--connect-timeout", str(HTTP_CONNECT_TIMEOUT),
             "--max-time", str(HTTP_MAX_TIME_UPLOAD),
@@ -134,7 +135,11 @@ def http_post_form(
                     return json.loads(result.stdout)
                 return {}
 
-            log_warn(f"HTTP POST form failed (attempt {attempt}/{max_retries}): curl exit {result.returncode}")
+            # Log curl exit code and stderr for better debugging
+            error_msg = f"curl exit {result.returncode}"
+            if result.stderr:
+                error_msg += f": {result.stderr.strip()}"
+            log_warn(f"HTTP POST form failed (attempt {attempt}/{max_retries}): {error_msg}")
             if attempt < max_retries:
                 time.sleep(1)
 
@@ -144,6 +149,9 @@ def http_post_form(
                 time.sleep(1)
         except json.JSONDecodeError as e:
             log_warn(f"HTTP POST form failed (attempt {attempt}/{max_retries}): Invalid JSON response")
+            # Log raw response for debugging (truncate to avoid log spam)
+            if result.stdout:
+                log_debug(f"Raw response: {result.stdout[:500]}")
             if attempt < max_retries:
                 time.sleep(1)
         except Exception as e:

--- a/turbo/apps/web/src/lib/e2b/scripts/lib/http_client.py.ts
+++ b/turbo/apps/web/src/lib/e2b/scripts/lib/http_client.py.ts
@@ -122,6 +122,7 @@ def http_post_form(
         if file_path:
             curl_cmd.extend(["-F", f"{file_field}=@{file_path}"])
 
+        result = None  # Initialize for use in except blocks
         try:
             result = subprocess.run(
                 curl_cmd,
@@ -148,9 +149,9 @@ def http_post_form(
             if attempt < max_retries:
                 time.sleep(1)
         except json.JSONDecodeError as e:
-            log_warn(f"HTTP POST form failed (attempt {attempt}/{max_retries}): Invalid JSON response")
+            log_warn(f"HTTP POST form failed (attempt {attempt}/{max_retries}): Invalid JSON response: {e}")
             # Log raw response for debugging (truncate to avoid log spam)
-            if result.stdout:
+            if result and result.stdout:
                 log_debug(f"Raw response: {result.stdout[:500]}")
             if attempt < max_retries:
                 time.sleep(1)

